situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java

```diff
@@ -40,31 +41,41 @@ public PipelineConfiguration(
         this.readBatchDelay = getReadBatchDelayFromConfiguration(delay);
     }
 
-    private PluginSetting getSourceFromConfiguration(final Map<String, Map<String, Object>> sourceConfiguration) {
-        final PluginSetting sourcePluginSetting = getFirstPluginSettingFromConfiguration(sourceConfiguration,
-                SOURCE_COMPONENT);
-        if (sourcePluginSetting == null) {
+    private PluginSetting getSourceFromConfiguration(final Map.Entry<String, Map<String, Object>> sourceConfiguration) {
+        if (sourceConfiguration == null) {
             throw new IllegalArgumentException("Invalid configuration, source is a required component");
         }
-        return sourcePluginSetting;
-    }
-
-    private PluginSetting getBufferFromConfiguration(final Map<String, Map<String, Object>> bufferConfiguration) {
-        return getFirstPluginSettingFromConfiguration(bufferConfiguration, BUFFER_COMPONENT);
+        return getPluginSettingFromConfiguration(sourceConfiguration);
     }
 
-    private List<PluginSetting> getProcessorsFromConfiguration(
-            final Map<String, Map<String, Object>> processorConfiguration) {
-        return getAllPluginSettingsFromConfiguration(processorConfiguration);
+    private PluginSetting getBufferFromConfiguration(final Map.Entry<String, Map<String, Object>> bufferConfiguration) {
+        if(bufferConfiguration==null) {
+            return null;
+        }
+        return getPluginSettingFromConfiguration(bufferConfiguration);
     }
 
     private List<PluginSetting> getSinksFromConfiguration(
-            final Map<String, Map<String, Object>> sinkConfiguration) {
-        final List<PluginSetting> sinkPluginSettings = getAllPluginSettingsFromConfiguration(sinkConfiguration);
-        if (sinkPluginSettings.isEmpty()) {
+            final List<Map.Entry<String, Map<String, Object>>> sinkConfigurations) {
+        if (sinkConfigurations==null || sinkConfigurations.isEmpty()) {
```

Comment:
nit: space - will report in checkstyle and will fail when we enforce them.

`sinkConfigurations == null || sinkConfigurations.isEmpty()`